src/qt/glcanvas.cpp

```diff
@@ -64,6 +64,259 @@ void wxQtGLWidget::paintGL()
     EmitEvent(event);
 }
 
+// ----------------------------------------------------------------------------
+// wxGLContextAttrs: OpenGL rendering context attributes
+// ----------------------------------------------------------------------------
+// GLX specific values
+
+wxGLContextAttrs& wxGLContextAttrs::CoreProfile()
+{
+//    AddAttribBits(GLX_CONTEXT_PROFILE_MASK_ARB,
+//                  GLX_CONTEXT_CORE_PROFILE_BIT_ARB);
+    SetNeedsARB();
+    return *this;
+}
+
+wxGLContextAttrs& wxGLContextAttrs::MajorVersion(int val)
+{
+    if ( val > 0 )
+    {
+        if ( val >= 3 )
+            SetNeedsARB();
+    }
+    return *this;
+}
+
+wxGLContextAttrs& wxGLContextAttrs::MinorVersion(int val)
+{
+    if ( val >= 0 )
+    {
+    }
+    return *this;
+}
+
+wxGLContextAttrs& wxGLContextAttrs::CompatibilityProfile()
+{
+    SetNeedsARB();
+    return *this;
+}
+
+wxGLContextAttrs& wxGLContextAttrs::ForwardCompatible()
+{
+    SetNeedsARB();
+    return *this;
+}
+
+wxGLContextAttrs& wxGLContextAttrs::ES2()
+{
+    SetNeedsARB();
+    return *this;
+}
+
+wxGLContextAttrs& wxGLContextAttrs::DebugCtx()
+{
+    SetNeedsARB();
+    return *this;
+}
+
+wxGLContextAttrs& wxGLContextAttrs::Robust()
+{
+    SetNeedsARB();
+    return *this;
+}
+
+wxGLContextAttrs& wxGLContextAttrs::NoResetNotify()
+{
+    SetNeedsARB();
+    return *this;
+}
+
+wxGLContextAttrs& wxGLContextAttrs::LoseOnReset()
+{
+    SetNeedsARB();
+    return *this;
+}
+
+wxGLContextAttrs& wxGLContextAttrs::ResetIsolation()
+{
+    SetNeedsARB();
+    return *this;
+}
+
+wxGLContextAttrs& wxGLContextAttrs::ReleaseFlush(int val)
+{
+    SetNeedsARB();
+    return *this;
+}
+
+wxGLContextAttrs& wxGLContextAttrs::PlatformDefaults()
+{
+    renderTypeRGBA = true;
+    return *this;
+}
+
+void wxGLContextAttrs::EndList()
+{
+//    AddAttribute(None);
+}
+
+// ----------------------------------------------------------------------------
+// wxGLAttributes: Visual/FBconfig attributes
+// ----------------------------------------------------------------------------
+// GLX specific values
+
+//   Different versions of GLX API use rather different attributes lists, see
+//   the following URLs:
+//
+//   - <= 1.2: http://www.opengl.org/sdk/docs/man/xhtml/glXChooseVisual.xml
+//   - >= 1.3: http://www.opengl.org/sdk/docs/man/xhtml/glXChooseFBConfig.xml
+//
+//   Notice in particular that
+//   - GLX_RGBA is boolean attribute in the old version of the API but a
+//     value of GLX_RENDER_TYPE in the new one
+//   - Boolean attributes such as GLX_DOUBLEBUFFER don't take values in the
+//     old version but must be followed by True or False in the new one.
+
+wxGLAttributes& wxGLAttributes::RGBA()
+{
+    return *this;
+}
+
+wxGLAttributes& wxGLAttributes::BufferSize(int val)
+{
+    if ( val >= 0 )
```

Comment:
IMO adding empty if blocks to not implemented functions does not make things any clearer. If you want to make something more obvious I suggest you write simple comments rather than empty code blocks.